File: core/src/main/java/com/starshootercity/abilities/custom/ToggleableAbility.java
package com.starshootercity.abilities.custom;

import com.starshootercity.abilities.Ability;

public interface ToggleableAbility extends Ability {
    default boolean shouldRegister() {
        ToggleableAbilities.registerAbility(this);
        return ToggleableAbilities.isEnabled(this);
    }
}

File: core/src/main/java/com/starshootercity/abilities/custom/ToggleableAbility.kt
package com.starshootercity.abilities.custom

import com.starshootercity.abilities.Ability
import com.starshootercity.abilities.custom.ToggleableAbilities.isEnabled
import com.starshootercity.abilities.custom.ToggleableAbilities.registerAbility

interface ToggleableAbility : Ability {
    fun shouldRegister(): Boolean {
        registerAbility(this)
        return isEnabled(this)
    }
}
